data-table/table.js

```diff
@@ -361,18 +368,104 @@ function createEntryCell(config, row, rowIndex, colIndex) {
                 option.innerHTML = config.datumConfig.values[fieldNum][i];
                 select.appendChild(option);
             }
-            label.appendChild(select);
+            field = select;
         } else {
             /**
              * FIXME: This error handling could be improved. Maybe a try-catch block?
              */
-            let errorText = "Cell field datatype not supported.";
-            throw errorText;
+            throw String("Cell field datatype not supported.");
         }
+
+        if (cellFieldHasCallback(config, fieldNum)) {
+            createCallbackListener(config, cell, field, fieldNum)
+        }
+
+        label.appendChild(field);
         cell.appendChild(label);
     }
 }
 
+/**
+ * Checks if a given field has a callback associated with it
+ * @param {object} config   - Table configuration object
+ * @param {Number} fieldNum - Index of the field within the cell
+ * @returns {boolean}       - Returns True if field has a callback
+ */
+function cellFieldHasCallback(config, fieldNum) {
+    return config.datumConfig.callbacks[fieldNum] !== undefined && config.datumConfig.callbacks[fieldNum] !== "None";
+}
+
+/**
+ * Creates a callback listener, distinguishing between the type of field to which the callback applies
+ * @param {object} config                               - Table configuration object
+ * @param {HTMLTableDataCellElement} cell               - Cell within the table
+ * @param {HTMLInputElement|HTMLSelectElement} field    - The field to which the callback applies
+ * @param {Number} fieldNum                             - Index of the field within the cell
+ * @returns {undefined}                                 - Doesn't return anything
+ */
+function createCallbackListener(config, cell, field, fieldNum) {
+
+    field.addEventListener("focusout", function () {
+        let fieldValue = null;
+
+        // Gets value of the field, depending on what type of field it is
+        if (field.classList.contains('cell-input')) {
+            fieldValue = [field.value.trim()]
+        } else if (field.classList.contains('cell-checkbox')) {
+            fieldValue = [field.checked]
+        } else if (field.classList.contains('cell-dropdown')) {
+            fieldValue = config.datumConfig.values[fieldNum][field.selectedIndex]
```

Comment:
Testing against `classList` feels error-prone - can you instead test against `type`? That's less likely to change over time.

Alternatively, do some OOP and create a `FieldBase`, `FieldNumber`, `FieldBool`, and `FieldDropdown` class, and have those classes handle all the logic of these if/else statements. You have several of these throughout the code, so doing this with OOP will make it much easier to add a new type in the future.

If you think that's a reasonable idea, it's probably outside of the scope of this ticket - add a JIRA ticket for it.

Reply:
Now testing against `type`.

Reasonable idea - making a note about that ticket.